mac/KeymanEngine4Mac/KeymanEngine4Mac/CoreWrapper/CoreHelper.m

```diff
@@ -56,12 +54,23 @@ -(unsigned long long) unicharStringLength:(unichar const *)string {
     return length;
 }
 
+/**
+ * Return the string length (number of unicode scalar, or UTF32, values) excluding the terminating null.
+ */
+-(unsigned long) scalarValueStringLength:(UTF32Char const *)string {
+    unsigned long length = 0lu;
+    if(NULL == string) return length;
+
+    while('\0' != string[length])
+        length++;
+
+    return length;
+}
```

Comment:
Tabbing should be 2-space, not 4 space. Also style guides says we should be surrounding loop with `{}`

```suggestion
-(unsigned long) scalarValueStringLength:(UTF32Char const *)string {
  unsigned long length = 0lu;
  if(NULL == string) {
    return length;
  }
  while('\0' != string[length]) {
    length++;
  }
  return length;
}
```